### PR TITLE
[docs][observability] O11y refactor 1/N

### PR DESCRIPTION
**Pull request by angelinalg**
## Why are these changes needed?

Cleaning up the Monitoring and Debugging section in phases. Improve discoverability by moving out contributor content, and creating a user guides section. This is the first phase that should not interfere with pending PRs and starts work on Troubleshooting structure at @rkooo567's request. cc: @rkooo567 @alanwguo @scottsun94 @emmyscode 

## Checks

- [x ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-contribute/profiling.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _ray-core-internal-profiling:
 
-Profiling (internal)
-====================
+Profiling
```
Comment (scottsun94) [inline doc/source/ray-contribute/profiling.rst:3]: Same comment as for "Debugging" Wondering if we should use a more explicit title.

**Review by scottsun94 — COMMENTED** [review]
LGTM. Left minor comments

File: doc/source/ray-contribute/debugging.rst [hunk 2]
```diff
@@ -1,5 +1,5 @@
-Debugging (internal)
-====================
+Debugging
```
[the remark this answers is not in the document]
Comment (angelinalg) [reply doc/source/ray-contribute/debugging.rst:1]: Good point! I'll clarify.

File: doc/source/ray-contribute/profiling.rst (hunk 1 above)
Comment (angelinalg) [reply doc/source/ray-contribute/profiling.rst:3]: ack!

**Review by richardliaw — APPROVED** [review]

**Comment by angelinalg** [comment]
@scottsun94 You can check out my edits to address your comments!

**Pull request merged**